src/web/tools/KeyEventManager.ts

```diff
@@ -0,0 +1,81 @@
+import { AdaptedEvent, EventTypes } from '../interfaces';
+import EventManager from './EventManager';
+import { PointerType } from '../../PointerType';
+
+export default class KeyEventManager extends EventManager<HTMLElement> {
+  private activationKeys = ['Enter', ' '];
+  private cancelationKeys = ['Tab'];
+  private isPressed = false;
+
+  private keyDownCallback = (event: KeyboardEvent): void => {
+    if (this.cancelationKeys.indexOf(event.key) !== -1 && this.isPressed) {
+      this.dispatchEvent(event, EventTypes.CANCEL);
+      return;
+    }
+
+    if (this.activationKeys.indexOf(event.key) === -1) {
+      return;
+    }
+
+    this.dispatchEvent(event, EventTypes.DOWN);
+  };
+
+  private keyUpCallback = (event: KeyboardEvent): void => {
+    if (this.activationKeys.indexOf(event.key) === -1 || !this.isPressed) {
+      return;
+    }
+
+    this.dispatchEvent(event, EventTypes.UP);
+  };
+
+  private dispatchEvent(event: KeyboardEvent, eventType: EventTypes) {
+    const adaptedEvent = this.mapEvent(event, eventType);
+    switch (eventType) {
+      case EventTypes.UP:
+        this.isPressed = false;
+        this.onPointerUp(adaptedEvent);
+        break;
+      case EventTypes.DOWN:
+        this.isPressed = true;
+        this.onPointerDown(adaptedEvent);
+        break;
+      case EventTypes.CANCEL:
+        this.isPressed = false;
+        this.onPointerCancel(adaptedEvent);
+        break;
+    }
+  }
+
+  public registerListeners(): void {
+    this.view.addEventListener('keydown', this.keyDownCallback);
+    this.view.addEventListener('keyup', this.keyUpCallback);
+  }
+
+  public unregisterListeners(): void {
+    this.view.addEventListener('keydown', this.keyDownCallback);
+    this.view.addEventListener('keyup', this.keyUpCallback);
+  }
+
+  protected mapEvent(
+    event: KeyboardEvent,
+    eventType: EventTypes
+  ): AdaptedEvent {
+    const viewRect = (event.target as HTMLElement)?.getBoundingClientRect?.();
+
+    const pointerPosition = {
+      x: viewRect.x + viewRect.width / 2,
+      y: viewRect.y + viewRect.height / 2,
+    };
+
+    return {
+      x: pointerPosition.x,
+      y: pointerPosition.y,
+      offsetX: pointerPosition.x,
+      offsetY: pointerPosition.y,
```

Comment:
`offsets` are measured with respect to `parent`, not `viewport`. I know it probably doesn't change much in case of keyboard, but I think we should stay consistent with that 😅 

```suggestion
      offsetX: viewRect.width / 2,
      offsetY: viewRect.height / 2,
```

Reply:
Good call, fixed this in [b76a0ef](https://github.com/software-mansion/react-native-gesture-handler/pull/3035/commits/b76a0ef94344932b1669b2dc7b0f5b6d5a813279)